### DISCUSSION TITLE
Label_prop.py memory requirement

### DISCUSSION THREAD

**Discussion by drchangliu** [snap-stanford/ogb#172] [MAG240M-LSC]
It is said that "256GB CPU memory required to run this script" for label_prop.py. We ran this on a machine with 256GB main memory and still ran out of memory. The OS used a little bit of memory but not much. There was no other process running on the machine. So did you ran this successfully on a machine with 256GB physical memory? Just confirming the memory requirement. Thanks.

Chang

**Comment by rusty1s** [comment]
Sorry for the inconveniences. The `label_prop.py` example indeed may consume slightly more memory than 256GB. I added a `--low-memory` option to the script that converts the one-hot labels to `torch.half`. That should be fine to run the script with 256GB of memory (consumes about 160GB of memory for me). The validation accuracy is nearly identical (58.43%).

**Reply by drchangliu** [reply]
The new version works. Thanks a lot!

**Reply by zbn123** [reply]
The following error occurs when I converts the one-hot labels to torch.half
![image](https://user-images.githubusercontent.com/22554360/118936175-e261a500-b97e-11eb-87eb-17cb3583d30c.png)

**Reply by zbn123** [reply]
How do you solve this problem

**Reply by weihua916** [reply]
Hmm, it works fine on my end:

```bash
python label_prop.py --low-memory

Namespace(alpha=0.9, low_memory=True, num_layers=3)
Reading adjacency matrix... Done! [250.81s]
Propagating labels... Done! [380.19s]
Train: 0.9909, Valid: 0.5843
```
